eras/shelley/impl/src/Cardano/Ledger/Shelley/PParams.hs

```diff
@@ -256,26 +257,36 @@ shelleyPParamsPairs ::
   PParamsHKD Identity era ->
   [a]
 shelleyPParamsPairs pp =
-  uncurry (.=) <$> shelleyPParamsHKDPairs (Proxy @Identity) pp
+  uncurry (.=)
+    <$> shelleyPParamsHKDPairs (Proxy @Identity) pp
+      ++ [ "collateralPercentage" .= Aeson.Null
+         , "costModels" .= Aeson.emptyObject
+         , "executionUnitPrices" .= Aeson.Null
+         , "maxBlockExecutionUnits" .= Aeson.Null
+         , "maxCollateralInputs" .= Aeson.Null
+         , "maxTxExecutionUnits" .= Aeson.Null
+         , "maxValueSize" .= Aeson.Null
+         , "utxoCostPerByte" .= Aeson.Null
+         ]
```

Comment:
Hey @Jimbo4350 I really don't think these are necessary and they certainly don't follow the ledger mantra of previous eras being unaware of features introduced in the future eras. With this approach we will be forced to add fields in all eras for every new parameter we will add in the future. Which is definitely wrong.

```suggestion
```

Reply:
I agree. The (justified) confusion stems from the definition of  `ProtocolParameters` in `cardano-api` is not parameterized on `era`. @palas we should favour the ledger's JSON serialization when it comes to which parameters are available in a given era.